### ISSUE TITLE
Display hierarchy of domains and micro-thesaurus

### ISSUE THREAD

**Issue by tfrancart** [NatLibFi/Skosmos#634]
We are working on an improvement in Skosmos to handle data structured in the following way :

```
@prefix skos: <http://www.w3.org/2004/02/skos/core#>
@prefix ex: <http://exemple.fr/>
@prefix dcterms: <http://purl.org/dc/terms/>

ex:thesaurus a skos:ConceptScheme ;
	skos:prefLabel "The Thesaurus"@en .

ex:mt1 a skos:ConceptScheme ;
	skos:prefLabel "Micro-Thesaurus 1"@en ;
	dcterms:subject ex:d1 .

ex:mt2 a skos:ConceptScheme ;
	skos:prefLabel "Micro-Thesaurus 2"@en ;
	dcterms:subject ex:d1 .

ex:mt3 a skos:ConceptScheme ;
	skos:prefLabel "Micro-Thesaurus 3"@en ;
	dcterms:subject ex:d2 .

### Begin Domains

ex:domains a skos:ConceptScheme ;
	skos:prefLabel "Special 'Domains' Concept Scheme"@en .

ex:d1 a skos:Concept ;
	skos:inScheme ex:domains ;
	skos:topConceptOf ex:domains ;
	skos:prefLabel "Domain 1"@en .

ex:d2 a skos:Concept ;
	skos:inScheme ex:domains ;
	skos:topConceptOf ex:domains ;
	skos:prefLabel "Domain 2"@en .

#### End Domains

ex:c1 a skos:Concept ;
	skos:prefLabel "Concept 1"@en ;
	skos:inScheme ex:thesaurus , ex:mt1 ;
	skos:topConceptOf ex:thesaurus , ex:mt1 ;
	skos:narrower ex:c1.1 .

ex:c1.1 a skos:Concept ;
	skos:prefLabel "Concept 1.1"@en ;
	skos:inScheme ex:thesaurus , ex:mt1 ;
	skos:broader ex:c1 .

ex:c2 a skos:Concept ;
	skos:prefLabel "Concept 2"@en ;
	skos:inScheme ex:thesaurus , ex:mt2 ;
	skos:topConceptOf ex:thesaurus , ex:mt2 .

ex:c3 a skos:Concept ;
	skos:prefLabel "Concept 3"@en ;
	skos:inScheme ex:thesaurus , ex:mt3 ;
	skos:topConceptOf ex:thesaurus , ex:mt3 .
```

The "domains" are top-level entries in the thesaurus. They are declared as skos:Concept used to "index" the micro-thesaurus with a dcterms:subject. This is typically how large thesaurus like Eurovoc are structured. See http://eurovoc.europa.eu/drupal/?q=download/subject_oriented&cl=en.
Using skos:Collections to represent this is also possible, but less optimal, mostly because :
1. This ConceptScheme-oriented approach is the one preferred in VocBench, so making Skosmos compatible with this would avoid transforming the data between VB and Skosmos
1. The Collection modelling does not allow to browse the full hierarchy starting from top-collections down to concept hierarchy (the concepts are listed flat under each collections)
1. Micro-thesaurus, as their name suggest, _are_ really small thesaurus, so theoretically best modelled as ConceptSchemes.

So, in order to do this, we will propose the following improvements :
1. Extend the REST API that returns the list of ConceptSchemes to also read optional "dcterms:subject" metadata, and include that in the JSON response;
1. Modify the way the tree is generated to generate top-level nodes for domains under which the micro-thesaurus will be inserted;
1. ConceptSchemes without subject information will be inserted as root of the hierarchy, as usual.

Also, we note that the hierarchy REST API make the assumption that a concept is topConceptOf one ConceptScheme only. We need to extend this to be able to properly open the navigation tree for every ConceptScheme in which a concept is declared. I will open a separate issue for this.

**Comment by osma** [comment]
As you probably know Skosmos already has (limited) support for vocabularies that consist of more than one ConceptScheme. This would be an extension of that. Contributions welcome!

**Comment by tfrancart** [comment]
Exactly yes. We have actually finished the development and we will propose a PR soon.

**Comment by osma** [comment]
This was implemented in PR #1113 (released in 2.10) but we forgot to close the issue. Closing it now.
